spec/unit/provider/service/init_spec.rb

```diff
@@ -83,6 +83,7 @@
       allow(provider_class).to receive(:defpath).and_return('tmp')
 
       @services = ['one', 'two', 'three', 'four', 'umountfs']
+      allow(Dir).to receive(:entries).and_call_original
```

Comment:
I was a bit worried why this surfaced just know. Seems that is reproducible only if you have the default external facts dir created(`$HOME/.facter/facts.d`). On `Facter.value`, Facter tries to load external facts so this means that a call to `Facter.value` goes to the system.

The call that is not stubbed is coming from `./lib/puppet/provider/service/init.rb:57:in `excludes'` so i believe a better fix would be to stub `Facter.value` in https://github.com/puppetlabs/puppet/blob/main/spec/unit/provider/service/init_spec.rb#L81-L82

```diff
@@ -80,6 +80,7 @@

   describe "when getting all service instances" do
     before :each do
+      allow(Facter).to receive(:value).with(:osfamily).and_return('redhat')
       allow(provider_class).to receive(:defpath).and_return('tmp')

       @services = ['one', 'two', 'three', 'four', 'umountfs']
```

Reply:
👍  sounds good, I'll take a look today